### PR TITLE
Don't delete unmodified fields in edit-profile submission

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers r? @chromakode (no rush, whenever)

When we load the Edit Profile view, its fields (bio, fullname, location) are populated by default from the userInfo object (result of identifying).  If we make changes, those changes live in the routeState object.

If you make a change to one field (e.g. location) and then submit your changes, the routeState object is used for *all* three fields, and so the ones you didn't change are reset to the empty string on the server, resulting in data loss of those fields.

### CODE REVIEW

File: shared/profile/edit-profile/index.js [hunk 1]
```diff
@@ -46,9 +47,9 @@ export default connect(
     return {
```
Comment (chrisnojima) [inline shared/profile/edit-profile/index.js]: How this works with the || is very problematic above. i ran into this when i was trying to make this use recompose and not use the store. We have a multiple sources of truth problem and really if want to keep this in the routeState we should have a way to load it once or something (ie if you go into it and delete your fullname it'll magically get replaced if this connect gets triggered, etc). So i think thats the root cause of this issue.

in terms of this code, you could also implement this entirely in the connect function and not change the onSumbit above. i'll make an example

Comment (chrisnojima) [inline shared/profile/edit-profile/index.js]:
```
class EditProfile extends Component<void, Props, void> {
  render () {
      // ...
      onEditProfile={this.props.onEditProfile} // why is this the same as onSumbit?
      onSubmit={this.props.onEditProfile}
    />
  }
}

// $FlowIssue type this connector
export default connect(
  (state, {routeState}) => {
      // ...
  },
  (dispatch, {routeState, setRouteState}) => {
    return {
      // ...
      onEditProfile: (bio, fullname, location) => dispatch(editProfile(bio, fullname, location)),
    }
  },
  (stateProps, dispatchProps, ownProps) => ({
      ...stateProps,
      ...dispatchProps,
      ...ownProps,
      onEditProfile: () => {dispatchProps(stateProps.bio, stateProps.funname, stateProps.location)},
  })
)(EditProfile)

```

**Comment by cjb** [comment]
@chrisnojima Thanks, I've switched to using your version now, it works.  PTAL?

But now I notice another problem, we're hitting CORE-3969 ("user card cached?").  @maxtaco says he'll probably promote that to this sprint.  Am guessing it won't need code changes on our end but I'll make them if it does.

**Comment by cjb** [comment]
@chrisnojima Thanks, removed the unnecessary onEditProfile render prop and flow type.

**Comment by keybase-ci-visdiff** [comment]
The commits 0d05a46afa42...60f1474109db introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **unlock_folders-Success** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485362230392-0d05a46afa42-60f1474109db-linux/unlock_folders-Success-0d05a46afa42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485362230392-0d05a46afa42-60f1474109db-linux/unlock_folders-Success-60f1474109db.png)  
   unlock_folders-Success comparison
 * Changed: **Tracker-Tracker_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485362230392-0d05a46afa42-60f1474109db-linux/Tracker-Tracker_Error-0d05a46afa42.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485362230392-0d05a46afa42-60f1474109db-linux/Tracker-Tracker_Error-60f1474109db.png)  
   Tracker-Tracker_Error comparison
</summary>

**Pull request merged**